plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineDirectManifest.kt

```diff
@@ -96,9 +83,46 @@ open class GenerateGramineDirectManifest @Inject constructor(
 
         if (buildType == BuildType.Simulation) {
             val simulationMrSigner = computeSigningKeyMeasurement().toHexString()
-            commands.add("-Dsimulation_mrsigner=$simulationMrSigner")
+            gramineManifestArgs.add("-Dsimulation_mrsigner=$simulationMrSigner")
+        }
+
+        /**
+         * In case of Python enclaves, we need to build them outside the conclave-build container.
+         * This is because Jep is installed in a user space, not a system space and will therefore produce different
+         * results if run inside or outside the container.
+         * All other enclaves are built in the container.
+         */
+        if (pythonEnclave.get()) {
+            val architecture = "x86_64-linux-gnu"
+            val ldPreload = executePython(
+                "from sysconfig import get_config_var; " +
+                        "print(get_config_var('LIBPL') + '/' + get_config_var('LDLIBRARY'))"
+            )
+            // The location displayed by 'pip3 show jep' is actually of the site/dist-packages dir, not the specific 'jep'
+            // dir within it. We assume this is the packages dir for other modules as well. If this assumption is
+            // incorrect then we'll need to come up with a better solution.
+            val pythonPackagesPath = commandWithOutput("pip3", "show", "jep")
+                .splitToSequence("\n")
+                .single { it.startsWith("Location: ") }
+                .substringAfter("Location: ")
+
+            gramineManifestArgs.add("-Darch_libdir=/lib/$architecture")
+            gramineManifestArgs.add("-Dld_preload=$ldPreload")
+            gramineManifestArgs.add("-Dpython_packages_path=$pythonPackagesPath")
+
+            commandLine(gramineManifestArgs)
+        } else {
+            // The location displayed by 'pip3 show jep' is actually of the site/dist-packages dir, not the specific 'jep'
+            // dir within it. We assume this is the packages dir for other modules as well. If this assumption is
+            // incorrect then we'll need to come up with a better solution.
+            val pythonPackagesPath = linuxExec.execWithOutput(listOf("pip3", "show", "jep"))
+                .splitToSequence("\n")
+                .single { it.startsWith("Location: ") }
+                .substringAfter("Location: ")
+
+            gramineManifestArgs.add("-Dpython_packages_path=$pythonPackagesPath")
+            linuxExec.exec(gramineManifestArgs)
```

Comment:
Why do we have this else block?

Reply:
Because we build gramine manifest in the docker for non-python enclave. In case of a python enclaves, we build the manifest outside of the enclave.

Reply:
I will clean up the comments. There are multiple ones and it's getting messy.

Reply:
I have updated the comments.